pkg/action/executor/move_util.go

```diff
@@ -633,57 +638,67 @@ func createClonePod(client *kclient.Clientset, pod *api.Pod,
 		"deployment",        // used by a replicationcontroller to adopt pod
 		"deploymentconfig",  // used by a replicationcontroller to adopt pod
 	}
-	copyPodWithoutSelectedLabels(pod, npod, excludeKeys)
 
-	// copy pod spec, annotations, and labels from the parent
-	kind := parent.GetKind()
-	parentName := parent.GetName()
-	// pod spec
-	podSpecUnstructured, found, err := unstructured.NestedFieldCopy(parent.Object, "spec", "template", "spec")
-	if err != nil || !found {
-		return nil, fmt.Errorf("movePod: error retrieving podSpec from %s %s: %v", kind, parentName, err)
-	}
+	copySpec := false
 	podSpec := api.PodSpec{}
-	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(podSpecUnstructured.(map[string]interface{}), &podSpec); err != nil {
-		return nil, fmt.Errorf("movePod: error converting unstructured pod spec to typed pod spec for %s %s: %v", kind, parentName, err)
-	}
-	// annotations
-	annotations := make(map[string]string)
-	annotationsUnstructured, found, err := unstructured.NestedFieldCopy(parent.Object, "spec", "template", "metadata", "annotations")
-	if err != nil {
-		return nil, fmt.Errorf("movePod: error retrieving annotations from %s %s: %v", kind, parentName, err)
-	}
-	if found {
-		if err := runtime.DefaultUnstructuredConverter.FromUnstructured(annotationsUnstructured.(map[string]interface{}), &annotations); err != nil {
-			return nil, fmt.Errorf("movePod: error converting unstructured annotations to typed annotations for %s %s: %v", kind, parentName, err)
+	if parent == nil {
+		// case of bare pod
+		copySpec = true
+	} else {
```

Comment:
Nit for your consideration: could it be a bit cleaner to default `copySpec` to `true` (bare pod), and only perform the else part `if parent != nil`?

Reply:
Updated!